python/paddle/fluid/optimizer.py

```diff
@@ -321,8 +339,15 @@ def minimize(self,
             optimize_ops = self._create_optimization_pass(params_grads, loss,
                                                           startup_program)
         else:
-            params_grads = append_backward(loss, parameter_list, no_grad_set,
-                                           [error_clip_callback])
+            if not self._backward_called:
+                self._backward(loss, startup_program, parameter_list,
+                               no_grad_set)
+
+            # optimize user specified parameters
+            if user_params_grads:
```

Comment:
take a look at tf.Optimizer.apply_gradients, you might have better idea

Reply:
Well the actual implementation is the same, though might need to seperate APIs in another PR would be more cleaner.